editor/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -306,18 +318,15 @@ class HierarchicalTermSelector extends Component {
 }
 
 export default compose( [
-	withAPIData( ( props ) => {
-		const { slug } = props;
-		return {
-			taxonomy: `/wp/v2/taxonomies/${ slug }?context=edit`,
-		};
-	} ),
-	withSelect( ( select, ownProps ) => {
+	withSelect( ( select, { slug } ) => {
 		const { getCurrentPost } = select( 'core/editor' );
+		const { getTaxonomy } = select( 'core' );
+		const taxonomy = getTaxonomy( slug );
 		return {
-			hasCreateAction: get( getCurrentPost(), [ '_links', 'wp:action-create-' + ownProps.restBase ], false ),
-			hasAssignAction: get( getCurrentPost(), [ '_links', 'wp:action-assign-' + ownProps.restBase ], false ),
-			terms: select( 'core/editor' ).getEditedPostAttribute( ownProps.restBase ),
+			hasCreateAction: taxonomy ? get( getCurrentPost(), [ '_links', 'wp:action-create-' + taxonomy.rest_base ], false ) : false,
+			hasAssignAction: taxonomy ? get( getCurrentPost(), [ '_links', 'wp:action-assign-' + taxonomy.rest_base ], false ) : false,
+			terms: taxonomy ? select( 'core/editor' ).getEditedPostAttribute( taxonomy.rest_base ) : [],
+			taxonomy,
```

Comment:
I echo my previous concerns regarding `taxonomy` and `rest_base`.

Reply:
I think I added a check for this specific component